bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,495 @@
+                    options.add(new StateOption(scene.getTargetID(), scene.getName()));
+                }
+
+                logger.debug("Setting {} options on bridge", options.size());
+
+                stateDescriptionProvider.setStateOptions(
+                        new ChannelUID(getThing().getUID(), SomfyMyLinkBindingConstants.CHANNEL_SCENES), options);
+
+                return response.getResult();
+            } else {
+                return new SomfyMyLinkScene[0];
+            }
+        }
+        catch (Exception e) {
+            logger.debug("Exception while getting scene list. Message: {}", e.getMessage());
+            return new SomfyMyLinkScene[0];
+        }
+    }
+
+    public void commandShadeUp(String targetId) throws SomfyMyLinkException {
+        try {
+            String command = buildShadeCommand("mylink.move.up", targetId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding shade up: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding shade up", e);
+        }
+    }
+
+    public void commandShadeDown(String targetId) throws SomfyMyLinkException {
+        try {
+            String command = buildShadeCommand("mylink.move.down", targetId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding shade down: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding shade down", e);
+        }
+    }
+
+    public void commandShadeStop(String targetId) throws SomfyMyLinkException {
+        try {
+            String command = buildShadeCommand("mylink.move.stop", targetId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding shade stop: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding shade stop", e);
+        }
+    }
+
+    public void commandScene(Integer sceneId) throws SomfyMyLinkException {
+        try {
+            String command = buildSceneCommand("mylink.scene.run", sceneId);
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding scene run: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding scene run", e);
+        }
+    }
+
+    private CompletableFuture<@Nullable Void> sendCommand(String command) {
+        CompletableFuture<@Nullable Void> future = new CompletableFuture<>();
+        commandExecutor.execute(() -> {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection(); OutputStream out = socket.getOutputStream()) {
+                    byte[] sendBuffer = command.getBytes(StandardCharsets.US_ASCII);
+                    // send the command
+                    out.write(sendBuffer, 0, sendBuffer.length);
```

Comment:
Make sure to flush after writing to output to make sure that it gets sent.
```suggestion
                    out.write(sendBuffer);
                    out.flush();
```

Reply:
done